src/core/config/index.js

```diff
@@ -0,0 +1,144 @@
+/**
+ * @prettier
+ */
+import deepExtend from "deep-extend"
+
+import ApisPreset from "core/presets/apis"
+import { parseSearch } from "core/utils"
+
+const defaultConfig = {
+  // eslint-disable-next-line camelcase
+  domNode: null,
+  spec: {},
+  url: "",
+  urls: null,
+  layout: "BaseLayout",
+  docExpansion: "list",
+  maxDisplayedTags: null,
+  filter: null,
+  validatorUrl: "https://validator.swagger.io/validator",
+  oauth2RedirectUrl: `${window.location.protocol}//${window.location.host}${window.location.pathname.substring(0, window.location.pathname.lastIndexOf("/"))}/oauth2-redirect.html`,
+  persistAuthorization: false,
+  configs: {},
+  custom: {},
+  displayOperationId: false,
+  displayRequestDuration: false,
+  deepLinking: false,
+  tryItOutEnabled: false,
+  requestInterceptor: (a) => a,
+  responseInterceptor: (a) => a,
+  showMutatedRequest: true,
+  defaultModelRendering: "example",
+  defaultModelExpandDepth: 1,
+  defaultModelsExpandDepth: 1,
+  showExtensions: false,
+  showCommonExtensions: false,
+  withCredentials: undefined,
+  requestSnippetsEnabled: false,
+  requestSnippets: {
+    generators: {
+      // eslint-disable-next-line camelcase
+      curl_bash: {
+        title: "cURL (bash)",
+        syntax: "bash",
+      },
+      // eslint-disable-next-line camelcase
+      curl_powershell: {
+        title: "cURL (PowerShell)",
+        syntax: "powershell",
+      },
+      // eslint-disable-next-line camelcase
+      curl_cmd: {
+        title: "cURL (CMD)",
+        syntax: "bash",
+      },
+    },
+    defaultExpanded: true,
+    languages: null, // e.g. only show curl bash = ["curl_bash"]
+  },
+  supportedSubmitMethods: [
+    "get",
+    "put",
+    "post",
+    "delete",
+    "options",
+    "head",
+    "patch",
+    "trace",
+  ],
+  queryConfigEnabled: false,
+
+  // Initial set of plugins ( TODO rename this, or refactor - we don't need presets _and_ plugins. Its just there for performance.
+  // Instead, we can compile the first plugin ( it can be a collection of plugins ), then batch the rest.
+  presets: [ApisPreset],
+
+  // Plugins; ( loaded after presets )
+  plugins: [],
+
+  pluginsOptions: {
+    // Behavior during plugin registration. Can be :
```

Comment:
Can we remove some of these comments for the options, as in this one and the ones for initialState, fn and requestSnippets.languages?

Reply:
I've removed comment for `initialState` in https://github.com/swagger-api/swagger-ui/pull/9815. But other comments still relay usable info. Why removing them?